dashboard-api/internal/platform/server/handler/studies/list.go

```diff
@@ -0,0 +1,113 @@
+package studies
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	studiesmanager "github.com/vlab-research/vlab/dashboard-api/internal"
+)
+
+type studyReponse struct {
+	Id        string `json:"id"`
+	Name      string `json:"name"`
+	Slug      string `json:"slug"`
+	CreatedAt int64  `json:"createdAt"`
+}
+
+func ListHandler(studyRepository studiesmanager.StudyRepository) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		pagination, err := parsePaginationQueryParams(ctx)
+		if err != nil {
+			ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+
+		studies, err := studyRepository.GetStudies(ctx, pagination.Cursor, pagination.Number)
+		if err != nil {
+			ctx.Status(http.StatusInternalServerError)
+			return
+		}
+
+		data := make([]studyReponse, 0, len(studies))
+		for _, study := range studies {
+			data = append(data, studyReponse{
+				Id:        study.ID(),
+				Name:      study.Name(),
+				Slug:      study.Slug(),
+				CreatedAt: study.CreatedAt(),
+			})
+		}
+
+		if len(studies) < pagination.Number {
```

Comment:
```Go
var nextCursor string
if len(studies) < pagination.Number {
   nextCursor = nil
} else {
   nextCursor = createNextCursor
}
```

Reply:
After finding that GO do not support ternary operator, the code that you provided was my first attempt but it didn't work out, because you cannot assign a **nil** to a **string variable.** That's the reason why I did it like this. (at the end, the repeated object is not that big as to impact readability, and extracting a function to returning the response and seems unnecessary)

A way to store a "value of a specific type" or "nil" in GO, is to use pointers. But for this particular scenario it will not work, because dereferencing a nil pointer will crash the application, so an extra conditional is required to check if the variable is a nil pointer. In case it's a nil pointer return it directly, otherwise dereference it.

Investigating further I found a way to solve the problem using and interface:
```
var nextCursor interface{} = nil
if len(studies) >= pagination.Number {
  nextCursor = createNextCursor(pagination)
}
```
(I know that interface zero value is nil, but I think it's more useful to make it explicit)